src/domain/Shift.ts

```diff
@@ -34,43 +35,37 @@ export default class Shift implements Comparable {
 	private allLessons: Lesson[]
 	private campus = ''
 	private occupation: ShiftOccupation
+	private classes: Class[]
 	
 	constructor(obj: ShiftDto, course: Course) {
 		this.course = course
 		this.name = obj.name
 
-		const fenixType = obj.types[0] as ShiftTypeFenix
-		// TODO: Improve, we should be able to get immediately
-		const type = Object.entries(ShiftTypeFenix).find(x => x[1] === fenixType)
-		const re = /^(.+)([\d]{2})$/
-		const match = this.name.match(re)
-		if (match === null || type === undefined) {
-			throw `Unexpected shift name - ${this.name}`
-		}
-		
-		this.type = Object.entries(ShiftType).filter(x => x[0] === type[0])[0][1]
-		this.shiftId = this.type + match[2]
+		const fenixType = obj.types[0]
 
-		if (obj.rooms !== null || (obj.rooms as string[]).length > 0) {
-			this.campus = obj.rooms[0]?.topLevelSpace.name
-		}
+		const type = Object.keys(ShiftTypeFenix).find(key => ShiftTypeFenix[key] === fenixType)
+		this.type = (type as string) as ShiftType
+		this.shiftId = this.name
+
+		this.campus = obj.classes[0].degree.campi[0].name
 
 		this.occupation = {
-			current: obj.occupation.current,
-			max: obj.occupation.max,
+			current: obj.enrolments.current,
+			max: obj.enrolments.maximum,
 		}
 
 		const lessons = obj.lessons.map((l: LessonDto) => {
+			const startDate = new Date(l.start)
+			const startTime = startDate.toLocaleTimeString()
+			const endTime = new Date(l.end).toLocaleTimeString()
```

Comment:
To allow for users in other timezones, perhaps we should convert the date to a local date on `Europe/Lisbon`?